src/elements/content-explorer/ContentExplorer.js

```diff
@@ -809,7 +809,8 @@ class ContentExplorer extends Component<Props, State> {
 
         const selectedItem: BoxItem = { ...item, selected: true };
 
-        this.updateCollection(currentCollection, selectedItem, () => {
+        const newItems = items.map(currentItem => (currentItem.id === item.id ? item : currentItem));
```

Comment:
Opted to create a helper function for updating an item in the collection instead of having `updateCollection` do too much since it's scope currently is to update the select state of all the items in the collection and call `setState`